edocumentation-domain/src/main/java/org/dcsa/edocumentation/domain/validations/AdvanceManifestFilingEBLValidator.java

```diff
@@ -0,0 +1,57 @@
+package org.dcsa.edocumentation.domain.validations;
+
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import org.dcsa.edocumentation.domain.persistence.entity.AdvanceManifestFilingEBL;
+
+public class AdvanceManifestFilingEBLValidator implements ConstraintValidator<AdvanceManifestFilingEBLValidation, AdvanceManifestFilingEBL> {
+
+  private static final CodeCache VALID_MANIFEST_CODES =
+    CodeCache.of("advancemanifestfilings.csv", "Advance Manifest Filing Type Code", "Country Code");
+
+  @Override
+  public boolean isValid(AdvanceManifestFilingEBL value, ConstraintValidatorContext context) {
+    if (value == null) {
+      return true;
+    }
+    context.disableDefaultConstraintViolation();
+    var state = ValidationState.of(value, context);
+    validateManifestTypeAndCountryCodes(state);
+    validateManifestFiling(state);
+    return state.isValid();
+  }
+
+  private void validateManifestTypeAndCountryCodes(ValidationState<AdvanceManifestFilingEBL> state) {
+    AdvanceManifestFilingEBL advanceManifestFilingEBL =  state.getValue();
+    if (!VALID_MANIFEST_CODES.isValid(advanceManifestFilingEBL.getManifestTypeCode(),advanceManifestFilingEBL.getCountryCode())) {
+      state.getContext().buildConstraintViolationWithTemplate("Mandatory to provide manifestTypeCode and countryCode for "
+          + advanceManifestFilingEBL.getShippingInstruction().getShippingInstructionReference())
+        // Match the TO path
+        .addPropertyNode("manifestTypeCode")
+        .addPropertyNode("countryCode")
```

Comment:
As I understand this code, it will claim the problem is at `.manifestTypeCode.countryCode` which is a non-existing property.

If so, it is better to have no property node (removing them both) and just use the `AdvanceManifestFilingEBL` as the problem (this is the default).

Reply:
I have removed and pushed it.